youtube_dl/jsinterp.py

```diff
@@ -18,13 +22,75 @@
     ('+', operator.add),
     ('%', operator.mod),
     ('/', operator.truediv),
-    ('*', operator.mul),
+    ('*', operator.mul)
 ]
 _ASSIGN_OPERATORS = [(op + '=', opfunc) for op, opfunc in _OPERATORS]
 _ASSIGN_OPERATORS.append(('=', lambda cur, right: right))
 
+_RESERVED_RE = r'(?:function|var|(?P<ret>return))\s'
```

Comment:
I've added test, but I don't see any problem.
And what do you mean @yan12125  "not context-free"?
Didn't you wanted to say not regular?